### ISSUE TITLE
Debugger throws exception

### ISSUE THREAD

**Issue by kdeltared** [DonJayamanne/pythonVSCode#135]
Using python 2.7.6 with Ubuntu 14.04.1. When trying to debug script with this line `import os, sys`

`Traceback (most recent call last):`
`File "/tmp/tmp-2784PJaJU4B0KtLx/visualstudio_py_debugger.py", line 1646, in loop cmd()`
`File "/tmp/tmp-2784PJaJU4B0KtLx/visualstudio_py_debugger.py", line 1918, in command_execute_code thread.run_on_thread(text, cur_frame, eid, frame_kind, repr_kind)`
`File "/tmp/tmp-2784PJaJU4B0KtLx/visualstudio_py_debugger.py", line 1246, in run_on_thread self.schedule_work(lambda : self.run_locally(text, cur_frame, execution_id, frame_kind, repr_kind))`
`File "/tmp/tmp-2784PJaJU4B0KtLx/visualstudio_py_debugger.py", line 1238, in schedule_work self.unblock()`
`File "/tmp/tmp-2784PJaJU4B0KtLx/visualstudio_py_debugger.py", line 1234, in unblock self._block_lock.release()`
`error: release unlocked lock`

**Comment by vchegwidden** [comment]
I have the same problem when debugging.. Only importing CppHeaderParser
Python 3.5.1 on Windows 7 x64
VsCode 1.1.1

```
Traceback (most recent call last):
File "C:\Users\me\AppData\Local\Temp\tmp-12708yjPl0ILUmoRZ\visualstudio_py_debugger.py", line 1646, in loop cmd()
File "C:\Users\me\AppData\Local\Temp\tmp-12708yjPl0ILUmoRZ\visualstudio_py_debugger.py", line 1918, in command_execute_code
thread.run_on_thread(text, cur_frame, eid, frame_kind, repr_kind)
File "C:\Users\me\AppData\Local\Temp\tmp-12708yjPl0ILUmoRZ\visualstudio_py_debugger.py", line 1246, in run_on_thread
self.schedule_work(lambda : self.run_locally(text, cur_frame, execution_id, frame_kind, repr_kind))
File "C:\Users\me\AppData\Local\Temp\tmp-12708yjPl0ILUmoRZ\visualstudio_py_debugger.py", line 1238, in schedule_work
self.unblock()
File "C:\Users\me\AppData\Local\Temp\tmp-12708yjPl0ILUmoRZ\visualstudio_py_debugger.py", line 1234, in unblock
self._block_lock.release()
RuntimeError: release unlocked lock
```

**Comment by vchegwidden** [comment]
Seems like the problem was caused by watch variables. When I deleted them I could debug as usual.
@kdeltared did you have any items in the watch window when the error occurred?

**Comment by DonJayamanne** [comment]
@kdeltared , are you still facing the same issue?

**Comment by juniorkrvl** [comment]
Hello guys! I'm having the same issue reported by @kdeltared, but like @vchegwidden said, when I remove the watched variables it works like a charm.

**Comment by DonJayamanne** [comment]
Great, i'll close this and create a trouble shooting section on the Wiki

**Comment by ConnectedReasoning** [comment]
I am having the same issue and I am running version 1.3 I have the same error messages so rather than creating a new issue I am adding to this one. It also works for me if I delete the watch variables. However I want to continue to use watch variables. Should this not be reopened?

**Comment by DonJayamanne** [comment]
What is the code you'd are debugging and what are the watch variables?

**Comment by basitanwer** [comment]
Put a watch on `number` and `ans`
On one of them it says `<error: previous evaluation has not completed>`
Keep debugging ultimately it throws the exception.

```
import numpy as np

n = 20; #maxValue = np.power(10,11
d = 7;

S = 0;

for number in range(4, n):
    max = number;
    ans = 1; #Everything is divisible by 1
    for i in range(2, max):
        if( max % i == 0):
            ans += i;
    if(ans % d == 0):
        S += ans;


print(S)
```

**Comment by kdeltared** [comment]
> vchegwidden commented on 25 May
Seems like the problem was caused by watch variables. When I deleted them I could debug as usual.
@kdeltared did you have any items in the watch window when the error occurred?
> 

Yes, deleting the watch variables works!

**Comment by DonJayamanne** [comment]
@basitanwer, I've tested the code you have provided and it doesn't cause the debugger to throw any exceptions.
- What is your OS?  
- What version of Python are you using? 
- At what point did you add the variables into the watch window (before or while debugging)?  

Thanks for your help.

**Comment by DonJayamanne** [comment]
@kdeltared, please could you provide me with some instructions to replicate this issue. 
Thanks, as you can understand a few people have reported this and I'd like to get to the bottom of this.

**Comment by ralexx** [comment]
@DonJayamanne, I can reproduce the `<error: previous evaluation has not completed>` part of @basitanwer's  issue in VSCode 1.31, Python plugin 0.3.18, Python 3.5.2, OSX 10.11.6. However no exceptions.

Is the error expected behavior?
![vscode_python_watch](https://cloud.githubusercontent.com/assets/6756310/17374145/17dda5aa-5960-11e6-9d00-1a92a819675d.png)

**Comment by DonJayamanne** [comment]
@ralexx , I too am able to reproduce the ```<error: previous evalu```, however that doesn't cause the debugger to crash. Thanks for your input.

**Comment by H-SG** [comment]
This is my code:

```python
import scipy.io

f = scipy.io.loadmat('/home/zander/Desktop/Melissa Data/training_images_test.mat')
data = f['training_image']
dataset = data[0,:]
data = None
del data
datalength = dataset.shape[0]
images = []
labels = []
print datalength
for x in range(0, datalength):
    images.append(dataset[x][0])
    labels.append(dataset[x][1])
```
I am watching `dataset` and `datalength`. A breakpoint exists at line 6 (`data = None`). Upon reaching the breakpoint I enter `test = dataset.shape[0]` into the debug console which causes the debugger to fail with behaviour I described in #240.

I'm creating a test .mat file which I can link to for you to use (after ensuring the error is still reproducible using it), my current one is far too large to share.

**Comment by H-SG** [comment]
I've tried it with a smaller .mat file, which works fine, the .mat file which I encouter the error with is 1.6GB in size. I think there might be an issue with watching `dataset` which can't load the amount of information in time and there is perhaps some thread timeout or similar issue?

Is there any other information I can provide?

**Comment by DonJayamanne** [comment]
@H-SG , @ralexx , @kdeltared, @vchegwidden , @juniorkrvl , @ConnectedReasoning , @basitanwer   unfortunately I haven't been able to replicate this very very elusive issue.
Would really appreciate it if someone else could assist here, either with an easier way to replicate or help in identifying the root cause.

**Comment by ralexx** [comment]
@DonJayamanne , I can't reproduce the crash either.

**Comment by DonJayamanne** [comment]
Will be closing until it can be replicated.

**Comment by DonJayamanne** [comment]
@ralexx , @basitanwer
>I can reproduce the ```error: previous evaluation has not completed``` part of @basitanwer's issue in VSCode 1.31, Python plugin 0.3.18, Python 3.5.2, OSX 10.11.6. However no exceptions.

This issue ```error: previous evaluation has not completed``` has been resolved in the new build.
